homeassistant/config_entries.py

```diff
@@ -341,6 +341,8 @@ def __init__(
 
         # Reason why config entry is in a failed state
         _setter(self, "reason", None)
+        self.reason_translation_key: str | None = None
+        self.reason_translation_placeholders: dict[str, str] | None = None
```

Comment:
Add these to `FROZEN_CONFIG_ENTRY_ATTRS` so they can't be set directly by integrations
Also, `reason_xxx` is not great, should we call the new attributes `error_reason_xxx` instead?